src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -978,9 +978,11 @@ def __call__(self, poller):
         except ClientException as client_exception:
             from azure.cli.core.commands.arm import handle_long_running_operation_exception
             self.cli_ctx.get_progress_controller().stop()
+            if getattr(client_exception, 'status_code', None) == 404 and 'delete' in self.cli_ctx.data['command']:
```

Comment:
it looks good to me in general  for long running operation delete. but sometimes delete will not be long running operation, and I am not if there is service returning 404 for delete succeed scenario. 
so for better understanding for further change, could we add comment for the hack?

Reply:
will you change influence the scenario where resource to delete doesn't exist?

Reply:
Good point. I will do some tests for the scenario.

Reply:
Tested and this logic is in the LongRunningOperation class and will only be called when service return a poller for LRO. If delete is not a LRO, the code will not reach here. 

For the case that the resource does not exist when delete, the service will return an error directly instead of a poller. So this change will not impact such case.

404 seems to me is a bug in service side. Let me talk to service team first and see if they can fix on their side.